File: code/1.two-sum.py
from typing import List

#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = []
        for i in range(len(nums)):
            n = target - nums[i]
            if n in l:
                return [l.index(n), i]
            else:
                l.append(nums[i])
        return [0, 0]


# @lc code=end

File: code/1.two-sum.rs
/*
 * @lc app=leetcode id=1 lang=rust
 *
 * [1] Two Sum
 */

// @lc code=start
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut acc: Vec<i32> = vec![];
        for i in 0..nums.len() {
            let n = target - nums[i];
            if acc.contains(&n) {
                return vec![acc.iter().position(|x| x == &n).unwrap() as i32, i as i32];
            } else {
                acc.push(nums[i])
            }
        }
        return vec![];
    }
}
// @lc code=end
